src/readii/process/images/crop.py

```
@@ -0,0 +1,449 @@
+    ------
+    ValueError
+        If the new dimensions are not valid for the image.
+    """
+    # Check that the number of dimensions in the new dimensions matches the number of dimensions in the image
+    if type(new_dimensions) == tuple:
+        if len(new_dimensions) != image.GetDimension():
+            msg = f"Number of dimensions in new_dimensions ({len(new_dimensions)}) does not match the number of dimensions in the image ({image.GetDimension()})."
+            logger.exception(msg)
+            raise ValueError(msg)
+
+    elif type(new_dimensions) == int:
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)